index.js

```diff
@@ -158,39 +159,48 @@ function _make_generateLog(metrics, func, start_time, config, context) {
           log('IOPIPE-DEBUG: ', JSON.stringify(response_body))
         }
 
-        var req = https.request({
-          hostname: config.url,
-          path: config.path,
-          port: 443,
-          method: 'POST',
-          headers: {'content-type' : 'application/json'},
-          agent: httpsAgent,
-          timeout: config.networkTimeout
-        }, (res) => {
-          var apiResponse = ''
-
-          res.on('data', function (chunk) {
-            apiResponse += chunk
-          })
+        dnsPromise.then((ipAddress) => {
+          var req = https.request({
+            hostname: ipAddress,
```

Comment:
ah, the comment says it resolves DNS on coldstarts, that's why I thought it did it that way